### DISCUSSION TITLE
prepared statement \"s70\" does not exist when using Accelerate

### DISCUSSION THREAD

**Discussion by sbkobaidze** [prisma/prisma#21894] [Q&A]
### Bug description

So I decided to use Accelerate for my Next.js project where I'm using Vercel's postgresql. So set-up is correct it was working at first but now It shows me this error and have no idea how to fix it. I tried adding pgBounce to Databse Direct URL params but still does not work.

### How to reproduce

Install accelerate and query info from vercel storage

### Prisma information

```prisma

// Add your code using Prisma Client
``
const session = await getServerSession(authOptions)
  const user = session
    ? await prisma.user.findUnique({
      include: {
        shipping: true,
        following:true,
        subscription: true,
        collections: true,
        affiliate:true,
        communities: true,
        likes:true,
       
          notified: {
            orderBy: {
              createdAt: "desc",
            },

            
            include: {
              product: true,
              notifier: true,
              
            },
            where: {
              read: false,
            },
          },
        },
        

        where: {
          email: session?.user?.email!,
      },

### Environment & setup

- OS: <!--[e.g. macOS, Windows, Debian, CentOS, ...]-->
- Database: <!--[PostgreSQL, MySQL, PlanetScale, MariaDB, SQL Server, SQLite, MongoDB or CockroachDB]-->
- Node.js version: <!--[Run `node -v` to see your Node.js version]-->

- Windows
- PostgreSQL
- 20.0

### Prisma Version

```
prisma 5.4.2
   "@prisma/client": "^5.5.2",

```

**Comment by janpio** [comment]
Do I assume correct that you are using the connection pooled connection string from Vercel Postgres (that includes `-pooler`)?
If so, you need to add `?pgbouncer=true` at the end of that connection string when configuring it in Prisma Accelerate.

As Accelerate is its own connection pooler, you can also just use the connection string that does _not_ include `-pooler` (and then also do not need to use `?pgbouncer=true`.

The `directUrl` value in your Prisma schema should point to a connection string that does _not_ include `-pooler` and should also not have `pgbouncer=true` as a parameter.

**Reply by sbkobaidze** [reply]
Yes I passed pgbouncer=true when configuring Accelerate, but now I just tried to update prisma to match prisma/client version and it works again. not sure if that was the solution or not since few hours ago error was gone automatically and came up again

**Reply by janpio** [reply]
(Note my edit to the message a few minutes later, I added the middle paragraph)

**Reply by sbkobaidze** [reply]
Even through everything is set-up as you mentioned it continues to show that error. So I decided to go back to vanilla prisma.

**Comment by benjamindell** [comment]
I'm also facing this same issue. So so frustrating. I am using Neon DB - which is basically the same platform that Vercel PostgreSQL uses. I am using the pooler postgresql URL and have pgbouncer=true.

Here's my connection string (with private info removed). Can anyone spot anything obvious i might be doing wrong?

`postgresql://USERNAME:PASSWORD@pooler-db-url.neon.tech:5432/neondb?sslmode=require&pgbouncer=true&connect_timeout=10&schema=public`

Essentially the issue is that almost every time i load the page i get an error like the below:

`ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(PostgresError { code: "26000", message: "prepared statement \"s20\" does not exist", severity: "ERROR", detail: None, column: None, hint: None }), transient: false })`

**Reply by benjamindell** [reply]
As a quick update, i am testing out what happens if I update the Accelerate DB URL to use the non-pooled version. So far its not erroring. I'll update if i see the error again.

**Reply by janpio** [reply]
The connection string you shared above looks correct. It has `pooler` in the hostname, and uses `pgbouncer=true`. You should not get this error if this is the database connection string used by Prisma to talk to the database.

If you are using Accelerate, this should be the connection string configured in there. The `prisma://` connection string should not be needed to be modified at all.

With Accelerate you can most probably also use the non pooled connection string indeed, and not bother with PgBouncer mode. Accelerate has similar benefits than using PgBouncer or some other connection pooler.

**Reply by benjamindell** [reply]
I removed the pooler and pgbouncer part from the connection string and it now works with Accelerate. Definitely issues when using pooler / pgbouncer with Accelerate, so it might be worth someone taking a look at some point as it was consistently throwing those errors on every page load.

Are there any downsides to me using the non pooler version with Accelerate?

**Reply by janpio** [reply]
> Are there any downsides to me using the non pooler version with Accelerate?

Generally no. Of course now every connection that Accelerate opens uses up 1 connection on your database directly. That changes the whole situation, but should not cause any problems in 99% of setups. 

(It can be a problem if you for example change your schma very, very often - which triggers Accelerate to start a new Prisma Query Engine under the hood that can open up to your conneciton limit's connections to the database. If you are using PgBouncer, that would be caught on that level. Without PgBouncer, that goes directly to the databasea nd could potentially exhaust your available connections.)

**Reply by benjamindell** [reply]
Gotcha. Are there any plans to get it working with pgbouncer? I'd rather stick with the pooled connection if I can, but its just too buggy right now.

**Reply by janpio** [reply]
It should be working with Accelerate actually, I have tests projects that do. 

Can you reproduce the problem in a new project in Accelerate?

**Reply by benjamindell** [reply]
Do you mean create a new / vanilla NextJS project and integrate with the existing Accelerate DB? Or create a new Accelerate project and link it to my existing NextJS project?

**Reply by janpio** [reply]
Optimally "new" for both options, but starting with a new accelerate project with the existing Next.js project would be a good start. Just making sure you are getting the same error again and it was not just a fluke.

**Reply by benjamindell** [reply]
Just setup a new Accelerate project (pointing to the same DB as before but with the pgbouncer / pool settings back in). And it worked for the first 5 or so refreshes of the page and then started throwing errors like before:

`ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(PostgresError { code: "26000", message: "prepared statement \"s45\" does not exist", severity: "ERROR", detail: None, column: None, hint: None }), transient: false })`

**Reply by janpio** [reply]
So PgBouncer 1.21.0, in transaction mode, with `max_prepared_statements` > 0, and `pgbouncer=true` in the connection string of that PgBouncer leads to this error message? Weird, that should not be possible at all.

**Reply by benjamindell** [reply]
Im not sure what version of  bgbouncer is being used. The database is hosted on Neon, with the following format:

`postgresql://USERNAME:PASSWORD@pooler-db-url.neon.tech:5432/neondb?sslmode=require&pgbouncer=true&connect_timeout=10&schema=public`

**Comment by kirtirajsinh** [comment]
I am getting the same Error when I try to query the Account from the Database. Some of the query don't throw this error. 

```
Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(PostgresError { code: "26000", message: "prepared statement \"s20\" does not exist", severity: "ERROR", detail: None, column: None, hint: None }), transient: false })

```
I have a 

DATABASE_URL = postgres://postgres.pqznpslkdrgvdwjvtmca:[YOUR-PASSWORD]@aws-0-ap-south-1.pooler.supabase.com:6543/postgres
DIRECT_URL = postgresql://postgres:[YOUR-PASSWORD]@db.pqznpslkdrgvdwjvtmca.supabase.co:5432/postgres

**Reply by UjjwalKaul** [reply]
I am facing the same error with the supabase url
Tried restarting database
nothing so far
do let me know if there is a solution

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
